packages/@aws-cdk/aws-lambda-event-sources/test/integ.kinesiswithdlq.ts

```diff
@@ -0,0 +1,64 @@
+import * as kinesis from '@aws-cdk/aws-kinesis';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as sqs from '@aws-cdk/aws-sqs';
+import { App, Stack } from "@aws-cdk/core";
+import { KinesisEventSource } from '../lib';
+
+/*
+ * Stack verification steps:
+ * * aws lambda invoke --function-name <deployed 'FP' function name> --invocation-type Event --payload '"Event"' response.json
+ * * Validate two executions of fn lambda
+ * * Validate one message in dlq
+ */
+
+// tslint:disable:no-console
+async function handler(event: any) {
+    console.log('event:', JSON.stringify(event, undefined, 2));
+    throw new Error();
+}
+
+class KinesisWithDLQTest extends Stack {
+    constructor(scope: App, id: string) {
+        super(scope, id);
+
+        const fn = new lambda.Function(this, 'F', {
+            runtime: lambda.Runtime.NODEJS_10_X,
+            handler: 'index.handler',
+            code: lambda.Code.fromInline(`exports.handler = ${handler.toString()}`)
+        });
+
+        const stream = new kinesis.Stream(this, 'S');
+
+        const dlq = new sqs.Queue(this, 'Q');
+
+        fn.addEventSource(new KinesisEventSource(stream, {
+            startingPosition: lambda.StartingPosition.TRIM_HORIZON,
+            onFailure: new lambda.SqsDLQ(dlq),
+            retryAttempts: 2
+        }));
+
+        new lambda.Function(this, 'FP', {
+            runtime: lambda.Runtime.NODEJS_10_X,
+            handler: 'index.handler',
+            code: lambda.Code.fromInline(`
+var AWS = require('aws-sdk');
+exports.handler = event => {
+    const kinesis = new AWS.Kinesis();
+    kinesis.putRecord({
+        Data: 'Hello World',
+        PartitionKey: 'Hello Key',
+        StreamName: '${stream.streamName}'
+    }, (err, data) => {
+        if (err) {
+            console.log(\`Error: \${err}\`);
+        }
+    })
+}
```

Comment:
Let's move this in a separate js or ts file and use `Code.fromAsset()`.

We've done something like that here - https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-apigateway/test/authorizers